_delphi_utils_python/delphi_utils/flash_eval/eval_day.py

```diff
@@ -0,0 +1,329 @@
+    streams_state = stream[list(filter(lambda x: len(x) == 2,
+                                       stream.columns))].unstack().dropna()  # .values
+    streams_groups.columns = streams_groups.columns.str[:2]
+    ranking_streams = {}
+    for key, group in streams_groups.stack().reset_index().groupby('level_1'):
+        for col, val in ts_streams.T.iterrows():
+            if key == col[:2]:
+                total_dist = pd.concat([group[0], streams_state]).reset_index(drop=True)
+                ranking_streams[col] = sum(total_dist < val[0]) / total_dist.shape[0]
+    stream_group = pd.Series(ranking_streams, name='stream_group')
+    return stream_group
+
+
+def setup_fips(flash_dir):
+    """Set up fips related dictionaries and population table.
+
+    Input: The directory location for files
+    Output: conversion dictionary state to fips & population per fips df
+    """
+    fips_lookup = pd.read_csv(f'{flash_dir}/fips.csv',
+                header=None).astype(str).set_axis(['FIPS', 'STATE'], axis=1)
+    fips_lookup['FIPS'] = fips_lookup['FIPS'].str.zfill(2)
+    fips_to_STATE = fips_lookup.set_index("FIPS").to_dict()['STATE']
+    STATE_to_fips = fips_lookup.set_index("STATE").to_dict()['FIPS']
+    fips = pd.read_csv(f'{flash_dir}/geo_fips.csv', index_col=0).set_index('geo')
+    orig_fips = fips[fips['pop'].isna()].index
+    geos_repl = [fips_to_STATE[x] for x in orig_fips.str[:2]]
+    fips.loc[orig_fips, 'pop'] = [float(fips[fips.index == x]['pop']) for x in geos_repl]
+    fips_pop_table = fips.unstack().to_frame().T
+    fips_pop_table.columns = [STATE_to_fips[x] if x in list(STATES)
+                          else x for x in fips_pop_table.columns.droplevel()]
+    return STATE_to_fips, fips_pop_table
+def flash_eval(lag, day, input_df, signal, params, logger=None):
+    """Evaluate most recent data using FlaSH.
+
+    Input:
+    lag: the difference between the reporting and reference date
+    day: the day of the reference date (today is the reporting date)
+    input_df: a df from the day for a particular signal that includes natl. state, and county data
+    params: additional params needed.
+    Ouput:
+    None
+    """
+    if not logger:
+        logger = get_structured_logger(
+            name=signal,
+            filename=params["common"].get("log_filename", None),
+            log_exceptions=params["common"].get("log_exceptions", True))
+
+    #TODOv4: Change these to a local dir
+    flash_dir = f'flash_ref/{signal}'
+    last_7 = pd.read_csv(f'{flash_dir}/last_7_{lag}.csv', index_col=0).astype(float)
+    wk_mean = pd.read_csv(f'{flash_dir}/weekday_mean_df_{lag}.csv', index_col=0)
+    wk_var = pd.read_csv(f'{flash_dir}/weekday_var_df_{lag}.csv', index_col=0)
+    weekday_params = pd.read_csv(f'{flash_dir}/weekday_params_{lag}.csv', index_col=0)
+    summary_stats = pd.read_csv(f'{flash_dir}/summary_stats_{lag}.csv', index_col=0)
+    stream = pd.read_csv(f'{flash_dir}/ret_df2_{lag}.csv', index_col=0)
```

Comment:
Is there some documentation for these reference files? How are they generated? Will there be a utility that generates these periodically?

Reply:
These files are generated from methods that aren't yet part of the FlaSH module here because we are still experimenting with the methods for file generation, but when we settle, we'll add those functions in + the logic that handles regeneration!